Read CSS as utf8 strings instead of binary

Jaifroid opened the pull request:
See discussion in #370 . It has yet to be determined whether JavaScript files should also be decoded as UTF8, or, since we will store them as Blobs, whether a binary format is appropriate for the browser to decode. Since we do not yet use the JavaScript function, that should be left to future investigation.

mossroy commented:
According to http://www.openzim.org/wiki/ZIM_file_format#Character_Encoding , everything should be encoded in UTF-8 inside ZIM files. @kelson42 : could you confirm it's also true for CSS and Javascript files?

mossroy commented:
@ernesst : the day after this PR will be merged, could you please test a nightly build on ubuntu touch? We're preparing a release of a version 2.3.

Jaifroid commented:
@mossroy , I've updated the tests.js, and tests are now passing.

Regarding JavaScript, as the most likely scenario is that we will be storing the scripts as Blobs, which are by definition binary format, and we'll then ask the browser to load them via the src attribute of the script tags, we should leave it to the browser to read/encode the JavaScript contents even in jQuery mode. I've tested this approach in the javaScript-support branch, and the scripts work fine that way.

So what I've done here is to approximate minimally the method used there. However, it won't actually work here, because with the innerHTML mode of injection, scripts attached by changing the src of a script tag do not read the files (by design). Therefore, the scripts will need to be attached by deleting the old script and creating a new script element. This is working in javaScript-support, but I feel it is a different issue and a different PR.

So let me know if you're happy with the skeleton I've put here. It's disabled in any case in the iframe onload section. I've added a note to point to #152 .

ernesst commented:
@mossroy sure no prob.

mossroy commented:
@kelson42 : yes, that's the wiki page I had read.
But when it says "both article data and URLs should be handled accordingly", it was not clear to me if "article data" was only the HTML page or also the javascript, CSS files that come with it.
Maybe this sentence could be more precise about this kind of content.

In any case, we now decode every textual content in UTF-8.

kelson42 commented:
@mossroy yes, the sentence is not so clear - but I have nothing more clear in mind. It just means that if the URL in the ZIM url table is encoded in UTF8 (it should be) then you better have an HTML/CSS page in UTF8 too if it points to it... because nothing will make a re-encoding on the fly if a link points from the HTML/CSS document to that resource.
